src/app/shared/components/grid-list/grid-list.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input, OnInit } from "@angular/core";
+
+@Component({
+    selector: "app-grid-list",
+    templateUrl: "./grid-list.component.html",
+    styleUrls: ["./grid-list.component.scss"]
+})
+export class GridListComponent implements OnInit {
+    @Input() mobileColumnCount: number = 1;
+    @Input() itemColumnSpan: number = 1;
+
+    constructor() { }
+
+    ngOnInit(): void {
+        this.mobileColumnCount = this.clampColumnValue(this.mobileColumnCount);
+        this.itemColumnSpan = this.clampColumnValue(this.itemColumnSpan);
+    }
+
+    getColumnClass(): string {
```

Comment:
if a method takes in no parameters and only derives value from another property, we should use class getters.